src/main/java/ch/saunah/saunahbackend/controller/SaunaController.java

```diff
@@ -59,6 +64,41 @@ ResponseEntity<String> editSauna(@RequestParam("Id") int id, @RequestBody SaunaT
         saunaService.editSauna(id, saunaTypeBody);
         return ResponseEntity.ok("success");
     }
+
+
+    @Operation(description = "Adds new images to the corresponding sauna.")
+    @PostMapping(value = "/sauna/{saunaId}/addImage", consumes = MediaType.MULTIPART_FORM_DATA_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public ResponseEntity<String> saveImages(@PathVariable("saunaId") int saunaId, @RequestBody List<MultipartFile> images)  {
+        saunaService.addSaunaImages(saunaId, images);
+        return ResponseEntity.ok("success");
+    }
+
+
+    @Operation(description = "Allows removing a existing Sauna with the ID specified.")
+    @PostMapping(path = "sauna/images/remove/{imageId}")
+    @ResponseBody
+    public ResponseEntity<String> removeImage(@RequestParam("imageId") int imageId) {
+        saunaService.removeSaunaImage(imageId);
+        return ResponseEntity.ok("success");
+    }
+
+    @Operation(description = "Returns the image file of the corresponding file name.")
+    @GetMapping(value = "/sauna/images/{fileName}")
+    @ResponseBody
+    public ResponseEntity<byte[]> getImage(@PathVariable("fileName") String fileName) throws IOException {
+        return saunaService.getImage(fileName);
+    }
+
+    @Operation(description = "Returns the image file of the corresponding file name.")
+    @GetMapping(value = "/sauna/{saunaId}/images")
+    @ResponseBody
+    public ResponseEntity<List<SaunaImageResponse>> getSaunaImages(@PathVariable("saunaId") int saunaId) {
+        List<SaunaImageResponse> saunaImages = saunaService.getSaunaImages(saunaId)
+            .stream().map(x -> new SaunaImageResponse(x.getId(), saunaId, x.getFileName())).collect(Collectors.toList());
+        return ResponseEntity.ok().body(saunaImages);
+    }
```

Comment:
Not quite clear what the difference is between these 2 methods. Make a more detailed description or better method names.